### ISSUE TITLE
IO / HTTP request overhead on KorGE default dispatcher

### ISSUE THREAD

**Issue by Josef-Vonasek** [korlibs/korio#126]
Issuing any http request on JVM has a ~20s delay.

For example, downloading a `3Kb` file with `UrlVfs` takes around ~25 seconds:
```
UrlVfs("https://raw.githubusercontent.com/korlibs/korio/master/README.md").readString()
```
Where 20s is waiting for the first byte!
Wrapping the request in a `runBlockingNoJs` seems to remove all of this overhead.

**Comment by soywiz** [comment]
Which target? JVM seemed to work fine for me

**Comment by Josef-Vonasek** [comment]
Yes. The overhead is still significant. I have updated the numbers.

**Comment by soywiz** [comment]
Do you have a small program reproducing the issue? Which dispatcher are you using. Is it the JVM?

```
suspend fun main() { yourcode.readString() }
```

reproduces the issue?

**Comment by Josef-Vonasek** [comment]
```kotlin
suspend fun main() = Korge(virtualWidth = 800, virtualHeight = 600, bgcolor = Colors["#2b2b2b"]) {
    println(UrlVfs("https://raw.githubusercontent.com/korlibs/korio/master/README.md").readString())
 ```

**Comment by soywiz** [comment]
Canyou try to wrap the readString within
```
withContext(Disptchers.IO) {}
```
The one you used is the default korge dispatcher and that might be the issue

**Comment by Josef-Vonasek** [comment]
`kotlinx.coroutines` defines only
```kotlin
public actual object Dispatchers {
    public actual val Default: CoroutineDispatcher = createDefaultDispatcher()
    public actual val Main: MainCoroutineDispatcher = JsMainDispatcher(Default, false)
    public actual val Unconfined: CoroutineDispatcher = kotlinx.coroutines.Unconfined
}
```

**Comment by soywiz** [comment]
I want to figure out what's the problem there

**Comment by soywiz** [comment]
The issue is that Korge/Korgw/GameWindow dispatcher uses a custom event loop dispatcher that is enqueuing executions.
And the problem is that the httpclient is performing a lot of suspensions while reading lines if the http headers and stuff, so it for example tries to read 3 bytes from the header, then it asycnhronously read the bytes and when the bytes arrive, it tries to continue the execution, but it has been enqueued for the next frame, not immediately so it has to wait 16 milliseconds even if it took microseconds to read those bytes.

To fix this, we should be processing events while waiting for vsync. The current implementation doesn't support this. But will work on a multithreaded renderer that waits for vsync in another thread and trigger an event when that happens. But that will take time.

**Comment by Josef-Vonasek** [comment]
This problem disappeared for me on Korge 2.3.1.
